File: leetcode_problems/typescript/src/lc455_assign_cokies.ts
function findContentChildren(g: number[], s: number[]): number {
  g.sort((a, b) => a - b);
  s.sort((a, b) => a - b);
  let p1 = 0;
  let p2 = 0;
  let res = 0;
  while (p1 < g.length && p2 < s.length) {
    if (g[p1] <= s[p2]) {
      p1++;
      p2++;
      res++;
    } else {
      p2++;
    }
  }
  return res;
}

File: leetcode_problems/cpp/lc455_assign_cokies.cpp
#include <vector>
#include <algorithm>
using std::vector;

class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int p1 = 0;
        int p2 = 0;
        int res = 0;
        while (p1 < g.size() && p2 < s.size()) {
            if (g[p1] <= s[p2]) {
                res++;
                p1++;
                p2++;
            } else {
                p2++;
            }
        }
        return res;
    }
};

int main() {
    vector<int> g = {1, 2, 3};
    vector<int> s = {1, 1};
    Solution sol;
    int result = sol.findContentChildren(g, s);
    std::cout << "Number of content children: " << result << std::endl;
    return 0;
}